### PR TITLE
Add stackblitz

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

* Add's an `open in stackblitz` button for applicable examples. Because we use named exports it is a bit more difficult to do.I wanted to open this up so people could give feedback before I add change a bunch of docs files.
* Some minor layout optimization for example code blocks

#### Description of how you validated changes

CleanShot 2022-06-15 at 12 30 02

| Before | After |
| - | - |
|  CleanShot 2022-06-15 at 12 26 49@2x |  CleanShot 2022-06-15 at 12 26 13@2x |
|  Simulator Screen Shot - iPhone 13 - 2022-06-15 at 12 22 10 |  Simulator Screen Shot - iPhone 13 - 2022-06-15 at 12 25 27 |
|  Simulator Screen Shot - iPhone 13 - 2022-06-15 at 12 35 23 |  Simulator Screen Shot - iPhone 13 - 2022-06-15 at 12 35 32 |

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/components/alert/react.mdx [hunk 1]
```diff
@@ -22,11 +22,11 @@ import {
...
 
 <Example>
-  <View>
+  <ExamplePreview>
     <DefaultAlertExample />
-  </View>
+  </ExamplePreview>
   
-  <ExampleCode>
+  <ExampleCode reactExport='DefaultAlertExample'>
```
Comment (dbanksdesign) [inline docs/src/pages/[platform]/components/alert/react.mdx:29]: We need the export name in order to generate code that imports it. See `openInStackblitz`

File: docs/src/styles/docs/main.scss [hunk 2]
```diff
@@ -14,7 +16,16 @@
 }
 
 .docs-content-body {
-  padding: var(--amplify-space-xxl);
+  padding: var(--amplify-space-medium);
```
Comment (dbanksdesign) [inline docs/src/styles/docs/main.scss:19]: Adjusting the padding to look better on smaller screens.

File: docs/src/utils/openInStackblitz.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+const rootElement = document.getElementById('root');
+const root = createRoot(rootElement);
+root.render(
+  <StrictMode>
+    <AmplifyProvider>
+      <${name} />
+    </AmplifyProvider>
+  </StrictMode>
+);
+`,
```
Comment (dbanksdesign) [inline docs/src/utils/openInStackblitz.ts:38]: This is a default index file for a CRA app that will import the component and render it based on the `name` passed to this function.

File: docs/src/utils/openInStackblitz.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  description = 'Amplify UI example',
+  name,
+  text,
+}: StackBlitzProps) => {
+  const fileName = `src/${name}.js`;
+
+  stackBlitz.openProject(
+    {
+      files: {
+        [fileName]: `import * as React from 'react';\n${text}`,
```
Comment (dbanksdesign) [inline docs/src/utils/openInStackblitz.ts:21]: Adding `import * as React from 'react';` to the file contents, should we just be adding this to our example code instead?

File: docs/src/styles/docs/example.scss [hunk 5]
```diff
@@ -50,9 +46,18 @@
...
-    background: var(--amplify-colors-background-secondary);
+    position: absolute;
     bottom: var(--amplify-space-small);
-    top: auto;
     right: var(--amplify-space-medium);
   }
 }
+
+.docs-example__preview,
+.docs-example > *:first-child {
```
Comment (dbanksdesign) [inline docs/src/styles/docs/example.scss:56]: This is to not break examples that don't use the new `ExamplePreview` component to wrap the preview.

File: docs/src/styles/docs/example.scss [hunk 6]
```diff
@@ -1,37 +1,33 @@
...
   height: 100%;
   padding: var(--amplify-space-relative-xxxl);
 }
 
-.example pre {
-  margin: 0;
-}
 
-.example-code {
+.docs-example__code {
```
Comment (dbanksdesign) [inline docs/src/styles/docs/example.scss:21]: Using BEM syntax

**Comment by jacoblogan** [comment]
Updated this draft PR to bring it up to date with main, but this feature is currently blocked on resolving some performance issues we are seeing with running amplify apps in stackblitz

**Pull request closed without merging**